### PR TITLE
feat(Bonus Pagamenti Digitali): [#176191528,#176269223] Supercashback graphical component and textual summary update

### PR DESCRIPTION
**Pull request by fabriziofff**
## Short description
This pr adds the Supercashback graphical component and update the textual summary conditions for the `Active` state.

```typescript
export const ActiveTextualSummary: React.FunctionComponent<Props> = props => {
  // active period but still not enough transaction
  if (
    props.period.amount.transactionNumber < props.period.minTransactionNumber &&
    props.period.amount.totalCashback > 0
  ) {
    return <Warning period={props.period} />;
  }
  if (
    props.period.amount.transactionNumber >= props.period.minTransactionNumber
  ) {
    // The user is in the supercashback ranking atm
    if (
      isBpdRankingReady(props.period.ranking) &&
      props.period.ranking.ranking <= props.period.minPosition
    ) {
      return (
        <SuperCashback superCashbackAmount={props.period.superCashbackAmount} />
      );
    }
    // The max cashback amount is reached
    {
      if (props.period.amount.totalCashback >= props.period.maxPeriodCashback) {
        return <MaxAmount name={props.name} />;
      }
    }
    // Cashback unlocked! visible for the next 10 transaction only
    if (
      props.period.amount.transactionNumber <=
      props.period.minTransactionNumber + 10
    ) {
      return <Unlock {...props} />;
    }
  }
  return null;
};

```

`Warning && transactionNumber==0`           |  `Warning && transactionNumber > 0` | `Unlock`
:-------------------------:|:------------:|:--------------------:
|   |  

`MaxAmount`           |  `SuperCashback` 
:-------------------------:|:--------------------:
|   

No Supercashback available   |
:----------------------------------------:
 |

## List of changes proposed in this pull request
- Added missing field `superCashbackAmount` in `BpdPeriod`
- Changed `BpdSummaryComponent` in order to display the supercashback ranking preview item `SuperCashbackRankingSummary`
- Deprecated `PercentageTransactionsSummary`
- Refactoring of `TextualSummary` and added new textual states
- Added `formatIntegerNumber` to format an integer with delimiter

## How to test
- run `ts/features/bonus/bpd/screens/details/components/summary/__test__/bpdSummaryComponent.test.tsx`
- run the application with different period / amount / ranking

### CODE REVIEW

File: ts/features/bonus/bpd/screens/details/components/summary/textualSummary/ActiveTextualSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    if (
+      isBpdRankingReady(props.period.ranking) &&
+      props.period.ranking.ranking <= props.period.minPosition
+    ) {
+      return (
+        <SuperCashback superCashbackAmount={props.period.superCashbackAmount} />
+      );
+    }
+    // The max cashback amount is reached
+    {
```
Comment (debiff) [inline ts/features/bonus/bpd/screens/details/components/summary/textualSummary/ActiveTextualSummary.tsx:105]: Why this curly brackets are needed?

**Review by debiff — APPROVED** [review]

File: ts/features/bonus/bpd/screens/details/components/summary/textualSummary/ActiveTextualSummary.tsx (hunk 1 above)
Comment (fabriziofff) [reply ts/features/bonus/bpd/screens/details/components/summary/textualSummary/ActiveTextualSummary.tsx]: removed  in 1b961af

File: ts/features/bonus/bpd/screens/details/components/summary/textualSummary/InactiveTextualSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import I18n from "../../../../../../../../i18n";
+import { dateToAccessibilityReadableFormat } from "../../../../../../../../utils/accessibility";
+import { BpdPeriod } from "../../../../../store/actions/periods";
+
+/**
+ * Inform the user about the start date of the next period
+ * @param props
+ * @constructor
+ */
+export const InactiveTextualSummary = (props: { period: BpdPeriod }) => (
```
[the remark this answers is not in the document]
Comment (fabriziofff) [reply ts/features/bonus/bpd/screens/details/components/summary/textualSummary/InactiveTextualSummary.tsx]: Added in 1b961af

File: ts/features/bonus/bpd/screens/details/components/summary/textualSummary/TextualSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  name: string | undefined;
+};
+
+/**
+ * Render additional text information for the user, related to the transactions and cashback amount
+ * Choose the textual infobox based on period and amount values
+ * @param props
+ * @constructor
+ */
+export const TextualSummary = (props: Props) => {
```
[the remark this answers is not in the document]
Comment (fabriziofff) [reply ts/features/bonus/bpd/screens/details/components/summary/textualSummary/TextualSummary.tsx]: Sure, added in 1b961af

File: ts/utils/stringBuilder.ts [hunk 4]
```diff
@@ -31,3 +31,15 @@ export const formatNumberCentsToAmount = (
...
+ * Format a number in a integer representation, removing all the decimal and adding the
+ * delimiter
+ * @param amount
+ */
+export const formatIntegerNumber = (amount: number): string =>
+  I18n.toNumber(amount, {
+    precision: 0,
+    delimiter: I18n.t("global.localization.delimiterSeparator"),
+    separator: I18n.t("global.localization.decimalSeparator")
+  });
```
[the remark this answers is not in the document]
Comment (fabriziofff) [reply ts/utils/stringBuilder.ts:45]: Added in e2b97cc

**Review by Undermaken — APPROVED** [review]

**Review by debiff — APPROVED** [review]

**Pull request merged**